File: db/mongo/gen_offset.rb

require 'mongo'
conn = Mongo::Connection.new("192.168.1.22")
conn.database_names
db = conn.db("dface")
coll = db.collection("offsets")

id = 1

File.open("offset.csv").each_line do |x| 
  arr = x.split(",").map{|i| i[1..-2].to_f}
  hash = { _id:id, loc: [arr[1],arr[0]], d:[arr[3],arr[2]]}
  coll.insert(hash)
  id+=1
end

=begin   

db.shops.find().forEach(function(x){
  var str = "gcj02_to_real(["+x.loc+"])";
  var loc2 = db.eval(str);
  db.shops.update({_id:x._id},{$set:{lo:loc2}});
})
 
=end

File: db/mongo/gen_offset.js
 
db.shops.find().forEach(function(x){
  if((typeof x.loc[0]) == "number"){
	  if( (typeof x.lo) == "object") return;
	  var str = "gcj02_to_real(["+x.loc+"])";
	  var loc2 = db.eval(str);
	  db.shops.update({_id:x._id},{$set:{lo:loc2}});
  }else{
	var arr = new Array( );
	for(var i=0;i<x.loc.length;i++){
	  var str = "gcj02_to_real(["+x.loc[i]+"])";
	  arr.push( db.eval(str) );	
	}
	db.shops.update({_id:x._id},{$set:{lo:arr}});
  }
  
})
